monkey/infection_monkey/monkey.py

```diff
@@ -156,10 +156,15 @@ def __init__(self, args, ipc_logger_queue: multiprocessing.Queue, log_path: Path
             / f"infection_monkey_plugins_{self._agent_id}_{secure_generate_random_string(n=20)}"
         )
         self._island_address, self._island_api_client = self._connect_to_island_api()
-        self._local_network_interfaces = get_network_interfaces()
         self._register_agent()
 
         self._operating_system = get_os()
+        self._local_network_interfaces = get_network_interfaces()
```

Comment:
`self._register_agent()` needs `self._local_network_interfaces`.
```suggestion
        self._local_network_interfaces = get_network_interfaces()
        self._register_agent()

        self._operating_system = get_os()
```